### PR TITLE
ug: Update the custom sota client doc

### PR DESCRIPTION
**Pull request by mike-sul**
Extend the custom sota client doc with info on new modes it supports.

# PR Template and Checklist

Please complete as much as possible to speed up the reviewing process.

Readiness and adding reviewers as appropriate is required.

All PRs should be reviewed by a technical writer/documentation team and a peer.
If effecting customers—which is a majority of content changes—a member of Customer Success must also review.

## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

* [x] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [x] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

File: source/user-guide/custom-sota-client.rst [hunk 1]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
 
 .. _custom-sota-client:
    https://github.com/foundriesio/meta-lmp/tree/main/meta-lmp-base/recipes-sota/custom-sota-client
+
+Custom SOTA Client Work Modes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+By default, the the example `SOTA client`_ works as a daemon updating a device to the latest version once it becomes available.
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:55]:
```suggestion
By default, the example `SOTA client`_ works as a daemon, updating a device to the latest version once it becomes available.
```

File: source/user-guide/custom-sota-client.rst [hunk 2]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
...
+In addition to the default daemon mode, users can run it as a CLI utility and perform specific steps of the overall update process separately.
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:56]:
```suggestion
In addition to the default daemon mode, users can run it as a CLI utility and perform specific steps of the update process separately.
```

File: source/user-guide/custom-sota-client.rst [hunk 3]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
...
+
+.. prompt:: bash
+
+    /build-custom/custom-sota-client --help
+    Usage:
+        custom-sota-client [cmd] [options]
+    Supported commands: check install run pull daemon
+    Default command is "daemon"
+
+* ``check`` - updates the device's TUF repo with the latest factory's TUF metadata or with the TUF metadata specified in the offline update bundle, and checks if there is a newer than currently installed target.
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:66]:
```suggestion
* ``check`` - updates the device's TUF repo with the latest Factory's TUF metadata or with the TUF metadata specified in the offline update bundle, and checks if there is a newer than currently installed Target.
```

File: source/user-guide/custom-sota-client.rst [hunk 4]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
...
+* ``pull`` - pulls the delta between the currently installed and the specified one.
+* ``install`` - installs the previously pulled target; yields an error if the specified target has not been pulled before.
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:68]:
```suggestion
* ``install`` - installs the previously pulled Target; yields an error if the specified Target has not been pulled before.
```

File: source/user-guide/custom-sota-client.rst [hunk 5]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
...
+* ``run`` - finalizes the installed target; confirms an update after reboot on a new rootfs version and/or starts the updated apps.
```
Comment (kprosise) [inline source/user-guide/custom-sota-client.rst:69]:
```suggestion
* ``run`` - finalizes the installed Target; confirms an update after reboot on a new rootfs version and/or starts the updated apps.
```

**Review by kprosise — APPROVED** [review]
Some small style related suggestions, otherwise it LGTM

**Comment by doanac** [comment]
Docs for 6888f6dc5f5fc44e124134ebd4ac720aa04e2fd2 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2613/docs/artifacts/html/index.html

**Comment by mike-sul** [comment]
@vanmaegima @caiotpereira @Tim-Anderson Please, let me know if this change makes sense to you.

**Review by MrCry0 — APPROVED** [review]

File: source/user-guide/custom-sota-client.rst [hunk 6]
```diff
@@ -49,3 +49,21 @@ Producing a factory-specific SOTA client can be done by:
...
+
+.. prompt:: bash
+
+    /build-custom/custom-sota-client --help
+    Usage:
+        custom-sota-client [cmd] [options]
+    Supported commands: check install run pull daemon
+    Default command is "daemon"
+
+* ``check`` - updates the device's TUF repo with the latest Factory's TUF metadata or with the TUF metadata specified in the offline update bundle, and checks if there is a newer than currently installed Target.
```
Comment (angolini) [inline source/user-guide/custom-sota-client.rst:66]: I understand "newer" in this sentence as a compatible Target which was created after the currently installed. The word "newer" is always tricky; that's why I'm double-checking.

**Review by angolini — APPROVED** [review]

File: source/user-guide/custom-sota-client.rst (hunk 6 above)
Comment (mike-sul) [reply source/user-guide/custom-sota-client.rst:66]: Your understanding is correct. Any suggestion to improve it is welcomed.

**Review by MrCry0 — APPROVED** [review]

**Comment by vanmaegima** [comment]
@mike-sul I like the additions, but I'm wondering what this means in terms of our support. Is this something we want our customers using rather than aktualizr-lite manual commands? Is this more reliable than aktualizr-lite manual commands? In our docs we state that we do not support manual aktualizr-lite, so it'd be good to understand the support scope here before merging. Adding @mike-scott to the discussion.

**Comment by mike-sul** [comment]
> Is this something we want our customers using rather than aktualizr-lite manual commands?

Not of course. This is just the API usage example.

>Is this more reliable than aktualizr-lite manual commands?

Maybe yes, maybe not, but it does not matter because this is just an example and we should not recommend using it as it is. The whole purpose of this example to provide customers with a guide on the API usage.  The aktualizr-lite manual commands (I guess you mean the CLI part) may be another API usage example, however its implementation is intermixed with the non-API daemon implementation. Therefore, I would not recommend to use it as an example.


> in our docs we state that we do not support manual aktualizr-lite, so it'd be good to understand the support scope here before merging. 

We support the API - this is our product we provide to customers. The example is one of the elements of the user documentation, guiding customers on the API usage.

How to implement a custom client is up to customers. It may be:

 * a daemon similar to our default update client (aklite daemon) that does strictly OTA functionality,
 * a more complex daemon that does many use-case specific stuff and triggers the API functions when its business logic decides that it needs to do some OTA-related functionality, e.g. check for update/new targets). (It reminds me Aeler case, they could implement their solution in such way),
 * a standalone CLI utility invoked by some other components.

**Comment by mike-sul** [comment]
@kprosise I suppose this PR is goof to go (be merged).

**Pull request merged**